upup/pkg/fi/cloudup/bootstrapchannelbuilder.go

```diff
@@ -58,9 +61,31 @@ func (b *BootstrapChannelBuilder) Build(c *fi.ModelBuilderContext) error {
 		Contents: fi.WrapResource(fi.NewBytesResource(addonsYAML)),
 	}
 
+	sv, err := api_util.ParseKubernetesVersion(b.cluster.Spec.KubernetesVersion)
+	if err != nil {
+		return fmt.Errorf("unable to determine kubernetes version from %q", b.cluster.Spec.KubernetesVersion)
+	}
+
+	optionsContext := &components.OptionsContext{
+		KubernetesVersion: *sv,
+	}
+
+	isVersionGTE1_6 := optionsContext.IsKubernetesGTE("1.6")
+	isVersionLT1_6 := optionsContext.IsKubernetesLT("1.6")
+
 	for key, manifest := range manifests {
 		name := b.cluster.ObjectMeta.Name + "-addons-" + key
 
+		if strings.Contains(manifest, "1.6") {
+			pre16 := strings.Contains(manifest, "pre-k8s-1.6")
+
+			if pre16 && isVersionGTE1_6 {
```

Comment:
We should use the actual manifest selection logic here, rather than relying on the key being set correctly